server/src/internalClusterTest/java/org/elasticsearch/search/fetch/subphase/highlight/HighlighterSearchIT.java

```diff
@@ -3495,6 +3501,392 @@ public void testDisableHighlightIdField() throws Exception {
         }
     }
 
+    public void testConstantKeywordFieldHighlightingQueryString() throws IOException {
+        String index = "test";
+        String constantKeywordFieldName = "test_constant_keyword_field";
+        String constantValue = "constant_value";
+
+        XContentBuilder mappings = prepareConstantKeywordMappings(constantKeywordFieldName, constantValue);
+        assertAcked(prepareCreate(index).setMapping(mappings));
+
+        XContentBuilder document = jsonBuilder().startObject()
+            .field("foo", "bar")
+            .field(constantKeywordFieldName, constantValue)
+            .endObject();
+        saveDocumentIntoIndex(index, "1", document);
+
+        SearchResponse search = prepareConstantKeywordSearch(QueryBuilders.queryStringQuery(constantValue));
+
+        assertNoFailures(search);
+        assertHighlight(search, 0, constantKeywordFieldName, 0, 1, equalTo("<em>constant_value</em>"));
+    }
+
+    public void testConstantKeywordFieldNoHighlightingQueryString() throws IOException {
+        String index = "test";
```

Comment:
I think these `NoHighlight` tests will be more useful if they return a hit, but then check that there isn't a highlight for the relevant field.  So this one could have a term query that matches `foo:bar` which will return the document, and then we can assert that there is no highlight on the constant field.

Reply:
Done!